ui/src/onboarding/containers/LoginPageContents.tsx

```diff
@@ -0,0 +1,383 @@
+      {
+        connection: Auth0Connection.Authentication,
+        email,
+        password,
+        family_name,
+        given_name,
+      },
+      err => {
+        if (err) {
+          this.displayErrorMessage(errors, err)
+          this.setState({buttonStatus: ComponentStatus.Default})
+          return
+        }
+        // log the user into Quartz
+        auth0.login(
+          {
+            realm: Auth0Connection.Authentication,
+            email,
+            password,
+          },
+          error => {
+            if (error) {
+              this.setState({buttonStatus: ComponentStatus.Default})
+              return this.displayErrorMessage(errors, error)
+            }
+          }
+        )
+      }
+    )
+  }
+
+  private displayErrorMessage = (errors, auth0Err) => {
+    if (auth0Err.code.includes('error in email')) {
+      this.setState({
+        ...errors,
+        emailError: 'Please enter a valid email address',
+      })
+    } else if (auth0Err.code === 'user_exists') {
+      const emailError = `An account with that email address already exists. Try logging in instead.`
+      this.setState({...errors, emailError})
+    } else {
+      const emailError = `We have been notified of an issue while creating your account. If this issue persists, please contact support@influxdata.com`
+      this.setState({...errors, emailError})
+      throw new Error(auth0Err.description)
+    }
+  }
+
+  private handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    this.setState({[e.target.name]: e.target.value})
+  }
+
+  private handleTabChange = (value: string) => {
+    this.setState({activeTab: value})
+  }
+
+  private handleSocialClick = (connection: Auth0Connection) => {
+    auth0.authorize({
+      connection,
+    })
+  }
+
+  private handleForgotPasswordClick = () => {
+    const {email} = this.state
+    if (!email) {
+      this.setState({emailError: 'Please enter a valid email address'})
+      return
+    }
+    auth0.changePassword(
+      {
+        email,
+        connection: Auth0Connection.Authentication,
+      },
+      (err, successMessage) => {
+        if (err) {
+          this.setState({emailError: err.message})
+          return
+        }
+        // notify user that change password email was sent successfully
+        // By default auth0 will send a success message even if the operation fails:
+        // https://community.auth0.com/t/auth0-changepassword-always-returns-ok-even-when-user-is-not-found/11081/8
+        this.setState({
+          emailError: '',
+          notificationText: `${successMessage}
+            If you haven't received an email, please ensure that the email you provided is correct.`,
+          notificationVisible: true,
+        })
+        // sets a time to remove the notification
+        setTimeout(() => {
+          this.setState({
+            notificationText: '',
+            notificationVisible: false,
+          })
+        }, 4000)
```

Comment:
why not use `shared/actions/notifications`, which handles all this for you?

Reply:
copy pasta'd too much 🤦‍♂